### PR TITLE
ncu-ci: include builtOn in data, update tests

### PR DESCRIPTION
**Pull request by joyeecheung**

### CODE REVIEW

File: test/fixtures/jenkins/build-failure-git-fetch/expected.json [hunk 1]
```diff
@@ -1,8 +0,0 @@
-[
```
Comment (joyeecheung) [inline test/fixtures/jenkins/build-failure-git-fetch/expected.json]: These are removed because they basically duplicate of `jenkins-failure-1`

**Comment by targos** [comment]
I'm getting an error when I try to run `ncu-ci` with this branch:

```
$ ncu-ci url https://ci.nodejs.org/job/node-test-pull-request-lite/881/
--------------------------------------------------------------------------------
Running LITE_PR: 881
--------------------------------------------------------------------------------
ncu-ci url <url>

Automatically detect CI type and show results

Positionals:
  url  URL of the PR or the CI                               [string] [required]

Options:
  --version  Show version number                                       [boolean]
  --copy     Write the results as markdown to clipboard         [default: false]
  --json     Write the results as json to the path                      [string]
  --help     Show help                                                 [boolean]

TypeError: Cannot read property 'display' of undefined
```

**Comment by joyeecheung** [comment]
@targos It does not work on master either, since currently `ncu-ci url` only supports

- PR URL with one of the following CI
- `node-test-pull-request` -> `PRBuild`
- `node-test-commit` -> `CommitBuild`
- `benchmark-node-micro-benchmarks` -> `BenchmarkRun`

See https://github.com/nodejs/node-core-utils/blob/01818205ffa34f93c3e114694972f3fe5548e92b/bin/ncu-ci#L116-L133

Would not be too hard to add a few switches and classes to support other CIs though

**Comment by targos** [comment]
Oh, sorry about that

**Pull request merged**